File: src/shared/api/sound.js
import { invoke } from '@tauri-apps/api/core'

// 播放指定音频文件
export async function playSound(path, volume = 0.5) {
  return await invoke('play_sound', { path, volume })
}

// 播放蜂鸣音
export async function playBeep(frequency = 800, durationMs = 100, volume = 0.5) {
  return await invoke('play_beep', { frequency, durationMs, volume })
}

// 播放复制音效
export async function playCopySound() {
  return await invoke('play_copy_sound')
}

// 播放粘贴音效
export async function playPasteSound() {
  return await invoke('play_paste_sound')
}

// 播放滚动音效
export async function playScrollSound() {
  return await invoke('play_scroll_sound')
}

File: src-tauri/src/commands/sound.rs
use crate::services::sound::{SoundPlayer, AppSounds};
use std::path::Path;

#[tauri::command]
pub fn play_sound(path: String, volume: f32) -> Result<(), String> {
    SoundPlayer::play(Path::new(&path), volume);
    Ok(())
}

#[tauri::command]
pub fn play_beep(frequency: f32, duration_ms: u64, volume: f32) -> Result<(), String> {
    SoundPlayer::play_beep(frequency, duration_ms, volume);
    Ok(())
}

#[tauri::command]
pub fn play_copy_sound() {
    AppSounds::play_copy();
}

#[tauri::command]
pub fn play_paste_sound() {
    AppSounds::play_paste();
}

#[tauri::command]
pub fn play_scroll_sound() {
    AppSounds::play_scroll();
}

